pyupgrade.py

```diff
@@ -1143,14 +1215,14 @@ def visit_Call(self, node):  # type: (ast.Call) -> None
             self.six_raises[_ast_to_offset(node)] = node
         elif (
                 not self._in_comp and
-                self._class_info_stack and
-                self._class_info_stack[-1].def_depth == 1 and
+                last_frame and
+                last_frame.def_depth == 1 and
                 isinstance(node.func, ast.Name) and
                 node.func.id == 'super' and
                 len(node.args) == 2 and
                 all(isinstance(arg, ast.Name) for arg in node.args) and
-                node.args[0].id == self._class_info_stack[-1].name and
-                node.args[1].id == self._class_info_stack[-1].first_arg_name
+                cast(ast.Name, node.args[0]).id == last_frame.name and
+                cast(ast.Name, node.args[1]).id == last_frame.first_arg_name
```

Comment:
instead of `cast`, could conver the cheeky `all(...)` into two `isinstance` calls

Reply:
sure